dpctl/tensor/_ctors.py

```diff
@@ -761,8 +761,20 @@ def full(
         )
     else:
         order = order[0].upper()
-    dpctl.utils.validate_usm_type(usm_type, allow_none=False)
+    dpctl.utils.validate_usm_type(usm_type, allow_none=True)
     sycl_queue = normalize_queue_device(sycl_queue=sycl_queue, device=device)
+
+    if isinstance(fill_value, (dpt.usm_ndarray, np.ndarray, tuple, list)):
+        X = dpt.asarray(
+            fill_value,
+            dtype=dtype,
+            device=device,
```

Comment:
```suggestion
```

Since `normalize_queue_device` has been called, we only need to use `sycl_queue` argument.